src/cdk-lib/bedrock/agents/agent-collaborator.ts

```diff
@@ -0,0 +1,175 @@
+/**
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import { CfnAgent } from 'aws-cdk-lib/aws-bedrock';
+import { IGrantable, Grant } from 'aws-cdk-lib/aws-iam';
+import { AgentDescriptor } from './agent-descriptor';
+
+/**
+ * Enum for collaborator's relay conversation history types.
+ */
+export enum AgentCollaboratorType {
+  /**
+     * Supervisor agent.
+     */
+  SUPERVISOR = 'SUPERVISOR',
+
+  /**
+     * Disabling collaboration.
+     */
+  DISABLED = 'DISABLED',
+
+  /**
+     * supervisor router.
+     */
+  SUPERVISOR_ROUTER = 'SUPERVISOR_ROUTER'
+}
+
+/**
+ * Enum for collaborator's relay conversation history types.
+ */
+export enum RelayConversationHistoryType {
+  /**
+     * Sending to the collaborator.
+     */
+  TO_COLLABORATOR = 'TO_COLLABORATOR',
+
+  /**
+     * Disabling relay of conversation history to the collaborator.
+     */
+  DISABLED = 'DISABLED'
+}
+
+/******************************************************************************
+ *                         PROPS - Agent Collaborator Class
+ *****************************************************************************/
+export interface AgentCollaboratorProps {
+/**
+     * Descriptor for the collaborating agent.
+     */
+  readonly agentDescriptor: AgentDescriptor;
```

Comment:
is Agent Descriptor really needed? Can't this just be an IAgentAlias? Maybe we can provide a more intuitive prop name as well